### PR TITLE
Update to ReadMe.md

### PR DESCRIPTION
**Pull request by joeliedtke**
I updated the ReadMe.md file with the following:

- Reviewed text for grammar and clarity
- Reviewed for usage of the Kubeflow Fairing name
- Revised the doc to try to help explain anything that I felt might be unclear.
- Revised the doc for structure, e.g. renaming the "Overview" section as "Features" and moving up towards the top.

Please let me know if you have any suggestions or requests. Also, please review the content I added for technical accuracy.

Once we are happy with this doc, please hold off on merging the PR so I can ask another writer to review the PR as well.
Thanks,
Joe

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/fairing/116)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,44 @@
...
-If you already have Docker and kubectl configured, you're all set! If you are using [Kubeflow](https://github.com/kubeflow/kubeflow), you can take advantage of advanced features.
+If you already have Docker and kubectl configured, you're all set! If you are using
+[Kubeflow](https://www.kubeflow.org/), you can take advantage of advanced features,
+such as using [TFJob](https://www.kubeflow.org/docs/components/tftraining/) to train
+a TensorFlow model.
+
+## Installing Kubeflow Fairing
 
-## Installing `fairing`
+To install Kubeflow Fairing from PyPI:
```
Comment (r2d4) [inline README.md]: nit: its technically not on PyPI since we're installing it from git

File: README.md [hunk 2]
```diff
@@ -1,31 +1,44 @@
-# Fairing
+# Kubeflow Fairing
 
-Easily train ML models on Kubernetes, directly from your python code, or a Jupyter notebook.
+Easily package, deploy, and run your machine learning (ML) training jobs on Kubernetes,
+directly from Python code or a Jupyter notebook. 
+
+## Features
+
+Kubeflow Fairing provides functions to seemlessly train models remotely on Kubernetes
```
Comment (r2d4) [inline README.md]: I think the pitch should be "build, train, and deploy from anywhere"

Comment (r2d4) [reply README.md]: cc @abhi-g

File: README.md [hunk 3]
```diff
@@ -1,31 +1,44 @@
...
+clusters.
+
+- Build images in seconds, without Docker
```
Comment (r2d4) [inline README.md]: I'm not sure what the highlighted feature set should be but maybe something like

* Easy API for data scientists to use, but extensible for advanced users
* Run the same workflow locally on your laptop, on a notebook, or in an automated pipeline
* Pluggable preprocessor, builder, and deployer architecture to interface with different environments

File: README.md (hunk 1 above)
Comment (hamedhsn) [reply README.md]: @r2d4 what is the timeline/plan to publish the first version on pypi?

Comment (r2d4) [reply README.md]: There's one there now but its old. Once we get integration testing infra up and running would be a good time for some release automation.

File: README.md (hunk 3 above)
Comment (joeliedtke) [reply README.md]: How about this:

Kubeflow Fairing simplifies the process of packaging, deploying, and running your ML training jobs remotely.

- Easily package your ML training job's Python code and Jupyter notebooks as container images.
- Deploy and run your ML training job on a variety of environments, including: Kubernetes clusters, Kubeflow, or Google Cloud ML Engine.
- Kubeflow Fairing's extensible preprocessor, builder, and deployer architecture provides a way for it to interface with different ML training environments.

File: README.md (hunk 2 above)
Comment (joeliedtke) [reply README.md]: We can't quite claim "from anywhere." Packaging, deploying, and running jobs is how I interpret what Fairing does. I like "build", but I feel like that term collides with model development.  Here are a couple revisions, let me know what you think:
 
Easily package, deploy, and run your machine learning (ML) training jobs remotely,
directly from Python code or a Jupyter notebook. 

Easily build, deploy, and run your machine learning (ML) training jobs remotely,
directly from Python code or a Jupyter notebook. 

Easily build, deploy, and run your machine learning (ML) training jobs at scale on several environments,
directly from Python code or a Jupyter notebook. 

Easily package, deploy, and run your machine learning (ML) training jobs at scale,
directly from Python code or a Jupyter notebook.

Comment (karthikv2k) [reply README.md]: "Easily build, train, and deploy your machine learning (ML) models to Kubeflow
directly from Python code or a Jupyter notebook." ?

Comment (abhi-g) [reply README.md]: @joeliedtke how about the one suggested by Karthik. The "Build/Train/Deploy" is how we have proposed the [CUJ] (http://bit.ly/cuj_train_deploy_notebook) which is on [Kubeflow Roadmap](https://github.com/kubeflow/kubeflow/blob/master/ROADMAP.md).  This has been reviewed by multiple users/customers over the course of the last quarter.

File: README.md [hunk 4]
```diff
@@ -1,31 +1,35 @@
...
 
-Easily train ML models on Kubernetes, directly from your python code, or a Jupyter notebook.
+Easily build, train, and deploy your machine learning (ML) training jobs on Kubeflow or Kubernetes,
+directly from Python code or a Jupyter notebook. 
 
 ## Requirements
 
+To use Kubeflow Fairing, you need:
+
 - a kubeconfig
```
Comment (karthikv2k) [inline README.md]: @r2d4 can we add more context (link to external doc) on what kubeconfig and docker config means?

Comment (r2d4) [reply README.md]: SGTM - opened an issue. Let's merge this first

Comment (karthikv2k) [reply README.md]: Sounds good.

File: README.md (hunk 1 above)
Comment (joeliedtke) [reply README.md]: Updated the PR to remove the reference to PyPI.

**Pull request merged**